### ISSUE TITLE
[IMPROVEMENT] Find a way to mock `template_renderer`

### ISSUE THREAD

**Issue by thealphadollar** [CCExtractor/sample-platform#305]
Please prefix your issue with one of the following: [BUG], [PROPOSAL], [QUESTION].

Sample platform commit (found at the bottom of each page) : **X.X.X**

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [ ] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [ ] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [ ] I have checked that the issue I'm posting isn't already reported.
- [ ] I have checked that the issue I'm posting isn't already solved and no duplicates exist in [closed issues](https://github.com/CCExtractor/sample-platform/issues?q=is%3Aissue+is%3Aclosed) and in [opened issues](https://github.com/CCExtractor/sample-platform/issues)
- [ ] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

**My familiarity with the project is as follows (check one, eg [X]):**

- [ ] I have never visited/used the platform.
- [ ] I have used the platform just a couple of times.
- [ ] I have used the platform extensively, but have not contributed previously.
- [ ] I am an active contributor to the platform.

---

Unittests for `mod_auth` were added in #302 but we were not able to mock `template_renderer` which decorates a lot of the functions and due to this the testing is not rigorous.

[Here](https://github.com/CCExtractor/sample-platform/pull/302/files#diff-d2996fc5542bbee85e1558205e1be811R619) is an example testcase which fails to mock `template_renderer`.

A way is required to mock the same and improve the testing.

**Comment by thealphadollar** [comment]
A way to bypass `login_required` decorator is to create a dummy user and  login with the same for the duration of the request.

**Comment by satyammittal** [comment]
@thealphadollar yes, there is already lot of tests bypass login_required. you can follow the same technique. https://github.com/CCExtractor/sample-platform/blob/master/tests/test_customized/TestControllers.py#L22

**Comment by thealphadollar** [comment]
@satyammittal Mocking it is also easy, I need to find a way to mock `template_renderer` :D

**Comment by cfsmp3** [comment]
Closing this issue as it has become stale and the workaround is sufficient.

**Context:**
- This issue is from June 2019 (~5.5 years ago)
- The `template_renderer` decorator mocking limitation was noted but didn't block development
- The `mod_auth` test file now has **45 test cases** providing good coverage
- The platform has been running reliably with the current test suite

**Current state:**
The test file still has the comment `# NOTE: Not able to mock template_renderer decorator` but the tests work around this limitation effectively. The decorator primarily affects template rendering, which can be tested through integration tests or by testing the underlying logic separately.

**Recommendation:**
If more rigorous unit testing of decorated functions is needed in the future, consider:
1. Refactoring to separate business logic from view decoration
2. Using Flask's test client (which is already done in most tests)
3. Modern mocking approaches with `unittest.mock.patch.object`

Given the age of this issue and that it hasn't blocked development, closing as "won't fix."
